nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProcessSession.java

```diff
@@ -3406,6 +3396,41 @@ private void checkpoint(final StandardProcessSession session, final List<Provena
             this.contentSizeOut += session.contentSizeOut;
         }
 
+        private <K, V> void mergeMaps(final Map<K, V> destination, final Map<K, V> toMerge, final BiFunction<? super V, ? super V, ? extends V> merger) {
+            if (toMerge == null) {
```

Comment:
not sure that null check against destination makes sense in the way this is being used. If `toMerge` is null, we can simply do nothing, because there is nothing to merge. But if the destination is null, we should not return without merging the values that are expected to be merged. It makes more sense IMO to throw a NPE in such a case. Is also worth noting that currently, all objects that could be passed in for `destination` are guaranteed non-null and if that were to change, the existing unit tests would fail.